### PR TITLE
Implementing Welcome and Dashboard UI

### PR DESCRIPTION
**Pull request by ted3x**
Imported custom view ArcView for custom shapes used in layouts.
Added welcome fragment, where user enters name
Added Dashboard fragment, where all quizs are listed with welcome message.
Implemented SharedPreferences object to handle user results getting/saving.
Implemented Navigation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,90 @@
+# Built application files
```
Comment (Edna001) [inline .gitignore:1]: ეს ფაილი არ უნდა იყოს აქ

File: app/src/main/java/com/c0d3in3/quizapp/SharedPreferencesHandler.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.c0d3in3.quizapp
+
+import android.content.Context
+import android.content.SharedPreferences
+
+object SharedPreferencesHandler {
+    private val sharedPref = App.context.getSharedPreferences(App.context.getString(R.string.preference_key), Context.MODE_PRIVATE)
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/quizapp/SharedPreferencesHandler.kt:7]: key ჯობია კონსტანტად გქონდეს

File: app/src/main/java/com/c0d3in3/quizapp/ui/quiz_dashboard/QuizDashboardViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.c0d3in3.quizapp.App
+import com.c0d3in3.quizapp.SharedPreferencesHandler
+
+class QuizDashboardViewModel : ViewModel() {
+
+    val welcomeMessage = MutableLiveData<String>()
+    private val lastEntryPoints : Int = SharedPreferencesHandler.getUserPoints(App.currentUsername)
+
+    init {
+        welcomeMessage.value = "გამარჯობა ${App.currentUsername}"
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/quizapp/ui/quiz_dashboard/QuizDashboardViewModel.kt:14]: ტექსტები strings_ში გადაიტანე

File: app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.welcome_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        startQuizButton.setOnClickListener {
+            val name = nameET.text.toString()
+            if(name.isBlank())  Toast.makeText(activity, "სახელი ველი ცარიელია!", Toast.LENGTH_SHORT).show()
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt:29]: ესეც

File: app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                val navController = Navigation.findNavController(view)
+                navController.navigate(R.id.action_welcomeFragment_to_quizDashboard)
+            }
+        }
+        super.onViewCreated(view, savedInstanceState)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        viewModel = ViewModelProvider(this).get(WelcomeViewModel::class.java)
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt:41]: ისევ shared view model გაქვს?

File: app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt (hunk 4 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt:29]: Changed

File: app/src/main/java/com/c0d3in3/quizapp/ui/quiz_dashboard/QuizDashboardViewModel.kt (hunk 3 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/quizapp/ui/quiz_dashboard/QuizDashboardViewModel.kt:14]: Changed

File: app/src/main/java/com/c0d3in3/quizapp/SharedPreferencesHandler.kt (hunk 2 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/quizapp/SharedPreferencesHandler.kt:7]: Changed

File: .gitignore (hunk 1 above)
Comment (ted3x) [reply .gitignore:1]: My bad, should have removed from staged files.

File: app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt (hunk 5 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/quizapp/ui/welcome/WelcomeFragment.kt:41]: Changed

**Review by DavitSoitashviliSpace — APPROVED** [review]

**Pull request merged**